website/content/en/pre-docs/AWS/constraints.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "Provisioning Configuration"
+linkTitle: "Provisioning"
+weight: 10
+---
+
+The [Provisioner CRD]({{< ref "../reference/provisioner-crd.md" >}}) provides two sections for constraining nodes. 
+
+- [`spec.requirements`](../reference/provisioner-crd#specrequirements)
+  - Cloud Provider Agnostic 
+  - Kubernetes Well Known Labels
+  - This section includues generally applicable constraints (zone, instance type) that each cloud provider is expected to implement. 
+  - Reference the Providioner CRD for more information. 
+- [`spec.provider`](#specprovider)
+  - Cloud Provider Specific
+  - This section defines constraints that are unique to AWS, such as SecurityGroups.
+  - Defined below. 
+
+## spec.provider
+
+cloud provider specific constrains
+
+[Review these fields in the code.](https://github.com/awslabs/karpenter/blob/main/pkg/cloudprovider/aws/apis/v1alpha1/provider.go#L33)
+
+### InstanceProfile
+An `InstanceProfile` is a region specific EC2 resource that is comprised of a reference to a single global IAM role.
+
+It is required, and specified by name. A suitable `KarpenterNodeRole` is created in the getting started guide.
+
+```
+spec:
+  provider:
+    instanceProfile: MyInstanceProfile
+```
+
+### LaunchTemplate
+
+A launch template is a set of configuration values sufficent for launching an EC2 instance (e.g., AMI, storage spec).
+
+A custom launch templay *may optionally* be specified by name. If none is specified, Karpenter will automatically create a launch template.
+
+Review the [Launch Template documentation](launch-templates.md) to learn how to create a custom one.
+
+```
+spec:
+  provider:
+    launchTemplate: MyLaunchTemplate
+```
+
+### SubnetSelector
+By default, Karpenter discovers subnets by tags. Alternatively, cluster subnets may be directly enumerated. 
+
+Subnets may be specified by AWS tag, or by name. Either approach supports wildcards. 
+
+Each instance gets *one* subnet *chosen* from this list.
+
+**Examples**
+
+Select all subnets with a specified tag:
+```
+spec:
+  provider:
+    subnetSelector:
+      kubernetes.io/cluster/MyCluster: '*'
+```
+
+Select subnets by name:
+```
+  subnetSelector:
+    Name: subnet-0fcd7006b3754e95e
+```
+
+Select subnets by an arbitary AWS tag key/value pair:
+```
+  subnetSelector:
+    MySubnetTag: value
+
+Select subnets using wildcards:
+```
+  subnetSelector:
+    Name: *public* 
+    MySubnetTag: '' # all resources with this tag
+
+```
+
+### SecurityGroupSelector
+
+Karpenter uses the EKS default security group, unless another is specified. The security group of an instance is comperable to a set of firewall rules.
+
+EKS creates at least two security groups, review the documentation for more info.

```

Comment:
Can you include a hyperlink to the documentation?

Reply:
good idea, done.